apiserver/facades/client/machinemanager/machinemanager.go

```diff
@@ -336,13 +336,39 @@ func (mm *MachineManagerAPI) UpgradeSeriesPrepare(args params.UpdateSeriesArg) (
 	if err := mm.check.ChangeAllowed(); err != nil {
 		return params.ErrorResult{}, err
 	}
-	err := mm.createUpgradeSeriesLock(args)
+	err := mm.updateSeriesPrepare(args)
 	if err != nil {
 		return params.ErrorResult{Error: common.ServerError(err)}, nil
 	}
 	return params.ErrorResult{}, nil
 }
 
+func (mm *MachineManagerAPI) updateSeriesPrepare(arg params.UpdateSeriesArg) error {
+	if arg.Series == "" {
+		return &params.Error{
+			Message: "series missing from args",
+			Code:    params.CodeBadRequest,
+		}
+	}
+	machineTag, err := names.ParseMachineTag(arg.Entity.Tag)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	machine, err := mm.st.Machine(machineTag.Id())
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if arg.Series == machine.Series() {
+		return errors.Errorf("%s is already running series %s", machineTag, arg.Series)
+	}
+	principals := machine.Principals()
+	if err = machine.VerifyUnitsSeries(principals, arg.Series, arg.Force); err != nil {
+		return errors.Trace(err)
+	}
+
+	return machine.CreateUpgradeSeriesLock()
```

Comment:
Is the concern that something could have been added between the time we verified the units series until the series lock is taken?
Its a fair point, but given that creating the document is also going to start other events being queued, we certainly want to do a pre check. It may be that we want to then also do a post check.

Reply:
If the VerifyUnitsSeries() fails, we don't want units to run the pre hook, which is triggered by the machine lock existing.  CreateUpgradeSeriesLock() should be the next step though, and defer unlock if err happens from there.  Updated code, and added a todo list for after the lock is taken.